core/rfb.js

```diff
@@ -1543,6 +1872,90 @@
             sock.flush();
         },
 
+        atenKeyEvent: function (sock, keysym, down) {
+            var ks = XK2HID[keysym];
```

Comment:
> I believe (but am not 100% positive) that the virtual USB device presented to the host by the iKVM server always reports itself as having a US keyboard layout.

All keyboards tend to do that because manufacturers are lazy. Hence no OS respects that information. :)

> I'm not sure what alternative would be better. We could send whatever scancode the browser gives us to the iKVM server and hope that the host OS interprets it correctly, but that seems dangerous.

That would be the correct approach. Otherwise noVNC would need to have a lookup table per keyboard layout.

> I don't have non-US-layout keyboards or non-US-locale servers to test this with, unfortunately.

What's printed on the keys doesn't matter. Just configure your local settings to something different. French is a favourite in finding bugs. :)